server/lib/routes/post-metrics.js

```diff
@@ -44,6 +50,7 @@ const NAVIGATION_TIMING_TYPE = INTEGER_TYPE.allow(null).required();
 const BODY_SCHEMA = {
   broker: STRING_TYPE.regex(BROKER_PATTERN).required(),
   context: STRING_TYPE.regex(CONTEXT_PATTERN).required(),
+  deviceId: STRING_TYPE.regex(DEVICE_ID_PATTERN).allow('none').required(),
```

Comment:
Can you make these two new fields optional for a train? We should be able to handle crossover for users that load the client side JS bundle on train-94, but report to train-95. Those users won't have these two new fields and their metrics will be rejected.